### ISSUE TITLE
DeprecationWarning with py3.7 in tests_py35

### ISSUE THREAD

**Issue by mimi1vx** [ronf/asyncssh#196]
```
    6s] + /usr/bin/python3 -m unittest discover -s tests_py35
[    8s] /home/abuild/rpmbuild/BUILD/asyncssh-1.16.0/asyncssh/agent.py:212: DeprecationWarning: 'with (yield from lock)' is deprecated use 'async with lock' instead
[    8s]   with (yield from self._lock):
[   11s] .../home/abuild/rpmbuild/BUILD/asyncssh-1.16.0/asyncssh/stream.py:453: DeprecationWarning: 'with (yield from lock)' is deprecated use 'async with lock' instead
[   11s]   with (yield from self._read_locks[datatype]):
[   16s] ......../home/abuild/rpmbuild/BUILD/asyncssh-1.16.0/asyncssh/stream.py:517: DeprecationWarning: 'with (yield from lock)' is deprecated use 'async with lock' instead
[   16s]   with (yield from self._read_locks[datatype]):
[   17s] .
[   17s] ----------------------------------------------------------------------
[   17s] Ran 12 tests in 9.795s
```
https://docs.python.org/3/library/asyncio-task.html#generator-based-coroutines

**Comment by ronf** [comment]
Yeah - I know about this deprecation warning, but the suggested replacement syntax is only available in Python 3.5 and later, and since it is a fundamental syntax change I can't use it in any code which still needs to run under Python 3.4. Until I drop support for 3.4, all code in the main asyncssh directory must continue to use the older "asyncio.coroutine" and "yield" syntax for everything and can't use the suggested "async with lock".

Python 3.4's official "end of life" is coming up in a little over a week (2019-03-16 according to https://devguide.python.org/#branchstatus). Once that happens, I may consider dropping support for it from future versions of asyncssh and switching over to Python 3.5+ syntax throughout (which should improve readability). However, until I do that, I've decided to live with this deprecation warning. According to the link above, the deprecated syntax won't be removed until Python 3.10, so there's plenty of time.

**Comment by ronf** [comment]
These deprecation warnings are now addressed in AsyncSSH 2.0.0, which has been migrated to the modern Python async/await syntax and now requires Python 3.6 or later. For more details (including some minor incompatibilities from AsyncSSH 1.x), see https://asyncssh.readthedocs.io/en/latest/changes.html.

For now, I'll be maintaining a parallel AsyncSSH 1.x branch for those running Python 3.4 or 3.5, but it'll contain mostly only bug fixes and not any significant new features. New feature work will be done only in AsyncSSH 2.x.
